create_derivatives/routines.py

```diff
@@ -128,8 +132,125 @@ def ocr_pdf(self):
 
 
 class ManifestMaker:
-    # TO DO: make manifests
-    pass
+
+    def __init__(self):
+        self.THUMBNAIL_HEIGHT = 200
+        self.THUMBNAIL_WIDTH = 200
+        self.server_url = settings.IMAGESERVER
+        self.resource_url = "{}/iiif/3/".format(self.server_url)
+        self.fac = ManifestFactory()
+        self.fac.set_base_prezi_uri("{}/manifests/".format(self.server_url))
+        self.fac.set_base_image_uri(self.resource_url)
+        self.fac.set_debug("error")
+        self.upgrader = Upgrader()
+
+    def run(self):
+        bags_with_manifests = []
+        for bag in Bag.objects.filter(process_status=Bag.PDF):
+            jp2_files = sorted([str(f) for f in matching_files(str(Path(bag.bag_path, "data", "JP2")))])
+            manifest_dir = str(Path(bag.bag_path, "data", "MANIFEST"))
+            self.fac.set_base_prezi_dir(str(manifest_dir))
+            self.create_manifest(jp2_files, manifest_dir, str(Path(bag.bag_path, "data", "JP2")), bag.bag_identifier, bag.as_data)
+            bag.process_status = Bag.MANIFESTS_CREATED
+            bag.save()
+            bags_with_manifests.append(bag.bag_identifier)
+        msg = "Manifests successfully created." if len(bags_with_manifests) else "No manifests created."
+        return msg, bags_with_manifests
+
+    def create_manifest(self, files, manifest_dir, image_dir, identifier,
+                        obj_data):
+        """Method that runs the other methods to build a manifest file and populate
+        it with information.
+
+        Args:
+            files (list): Files to iterate over
+            manifest_dir (str): Path to directory to write manifests to.
+            image_dir (str): Path to directory containing derivative image files.
+            identifier (str): A unique identifier.
+            obj_data (dict): Data about the archival object.
+        """
+        manifest_path = "{}.json".format(str(Path(manifest_dir, identifier)))
+        page_number = 1
+        manifest = self.fac.manifest(ident=identifier, label=obj_data["title"])
+        cleaned_id = manifest.id[:-5]
```

Comment:
Basically removing the `.json` from the end of the id, which is automatically added by the prezi factory as far as I can tell.

Reply:
I haven't found a way to overwrite that functionality so far, so while it's not ideal, it works.

Reply:
OK, can we do something a little bit more obvious like calling `Path(manifest.id).stem`? I'm not a huge fan of list slicing by opaque indexes because it's not self-documenting...

Reply:
Also, wait, why are we pulling the `.json` off of the Manifest identifier?

Reply:
I was under the impression that it was something we wanted. I'm happy to leave it though.

I think it's acceptable if I'm reading [this](https://iiif.io/api/presentation/3.0/#52-manifest) correctly.

Reply:
The example we were talking about and I was looking at also has the `.json` removed: (https://iiif.harvardartmuseums.org/manifests/object/299843)

Reply:
@helrond Unfortunately, the manifest id is a URL, and not a filepath, so stem doesn't work that well. I could probably do something with urlparse, but I'm not sure that's desirable? 

We'll have to do some manipulation for removing the `.json` from the canvas id no matter what. Otherwise it'll show up in the middle of the id.

Reply:
OK, yeah looking at that example I get it now, thanks!

So looking at the iiif_prezi code it seems like if you pass a fully qualified URL in as the `ident` to `fac.manifest` you'll have the desired result. I've tested this locally and it seems to work.